cmd/ctl/check_config.go

```diff
@@ -0,0 +1,376 @@
+		Run: func(cmd *cobra.Command, args []string) {
+			if strings.TrimSpace(globalOpts.principalNamespace) == "" {
+				cmdutil.Fatal("--principal-namespace is required")
+			}
+			ctx := context.TODO()
+			clt, err := kube.NewKubernetesClientFromConfig(ctx, globalOpts.principalNamespace, "", globalOpts.principalContext)
+			if err != nil {
+				cmdutil.Fatal("Could not create Kubernetes client: %v", err)
+			}
+			results := RunPrincipalChecks(ctx, clt, globalOpts.principalNamespace)
+			printResultsAndExit(results)
+		},
+	}
+	return command
+}
+
+func NewCheckConfigAgentCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "agent",
+		Short: "Validate agent configuration (and principal cross-checks)",
+		Run: func(cmd *cobra.Command, args []string) {
+			if strings.TrimSpace(globalOpts.agentContext) == "" ||
+				strings.TrimSpace(globalOpts.agentNamespace) == "" ||
+				strings.TrimSpace(globalOpts.principalContext) == "" ||
+				strings.TrimSpace(globalOpts.principalNamespace) == "" {
+				cmdutil.Fatal("--agent-context, --agent-namespace, --principal-context, --principal-namespace are all required")
+			}
+			ctx := context.TODO()
+			agentClt, err := kube.NewKubernetesClientFromConfig(ctx, globalOpts.agentNamespace, "", globalOpts.agentContext)
+			if err != nil {
+				cmdutil.Fatal("Could not create agent Kubernetes client: %v", err)
+			}
+			principalClt, err := kube.NewKubernetesClientFromConfig(ctx, globalOpts.principalNamespace, "", globalOpts.principalContext)
+			if err != nil {
+				cmdutil.Fatal("Could not create principal Kubernetes client: %v", err)
+			}
+			// Run principal checks as part of agent checks
+			results := []checkResult{}
+			results = append(results, RunPrincipalChecks(ctx, principalClt, globalOpts.principalNamespace)...)
+			results = append(results, RunAgentChecks(ctx, agentClt.Clientset, globalOpts.agentNamespace, principalClt.Clientset, globalOpts.principalNamespace)...)
+			printResultsAndExit(results)
+		},
+	}
+	return command
+}
+
+func printResultsAndExit(results []checkResult) {
+	hasErr := false
+	fmt.Println("Configuration validation results:")
+	for _, r := range results {
+		fmt.Println(r.String())
+		if r.err != nil {
+			hasErr = true
+		}
+	}
+	if hasErr {
+		cmdutil.Fatal("one or more checks failed")
+	}
+}
+
+// RunPrincipalChecks validates the principal installation and related security assets.
+func RunPrincipalChecks(ctx context.Context, kubeClient *kube.KubernetesClient, principalNS string) []checkResult {
+	out := []checkResult{}
+
+	// Ensure Argo CD is cluster-scoped
+	out = append(out, checkResult{
+		name: "Verifying Argo CD is cluster-scoped",
+		err:  verifyArgoCDClusterScoped(ctx, kubeClient, principalNS),
+	})
+
+	// CA secret exists in principal namespace and is a valid TLS secret
+	out = append(out, checkResult{
+		name: fmt.Sprintf("Verifying principal public CA certificate exists and is valid (%s/%s)", principalNS, config.SecretNamePrincipalCA),
```

Comment:
JGW-TODO: What if name is different, for example `cr.Spec.ArgoCDAgent.Principal.TLS.RootCASecretName` points to a different value?